Prefer using commit identifier to build commit URL.

dewei-zhu opened the pull request:
#### fdc8cfeaf17e461bac1e48fb14db51a702bcd413
<pre>
Prefer using commit identifier to build commit URL.
<a href="https://bugs.webkit.org/show_bug.cgi?id=241646">https://bugs.webkit.org/show_bug.cgi?id=241646</a>
rdar://93104485

Reviewed by Jonathan Bedard.

Use commit identifier to construct commit URL and blame URL when available.
This simplifies URL construction for Git transition.

* Websites/perf.webkit.org/public/v3/models/commit-log.js: Added code to prefer using commit identifiers
when available.
(CommitLog.prototype.url):
(CommitLog.prototype.diff):
* Websites/perf.webkit.org/unit-tests/commit-log-tests.js: Added a unit test for diff.
(assert.deepStrictEqual.webkitGitCommitWithRevisionIdentifier.diff.oldWebKitGitCommitWithRevisionIdentifier):
* Websites/perf.webkit.org/unit-tests/commit-set-tests.js: Updated unit test.
* Websites/perf.webkit.org/unit-tests/resources/mock-v3-models.js: Added &apos;blameUrl&apos; to &apos;WebKitGit&apos; mock repository.
(MockModels.inject):

Canonical link: <a href="https://commits.webkit.org/251616@main">https://commits.webkit.org/251616@main</a>
git-svn-id: <a href="https://svn.webkit.org/repository/webkit/trunk@295611">https://svn.webkit.org/repository/webkit/trunk@295611</a> 268f45cc-cd09-0410-ab3c-d52691b4dbfc
</pre>

JonWBedard commented on Websites/perf.webkit.org/public/v3/models/commit-log.js:
```diff
@@ -42,7 +42,7 @@ class CommitLog extends DataModelObject {
     revision() { return this._rawData['revision']; }
```
Wonder if this function should return an r-prefixed string if the revision is an integer. Alternatively, could do that in `urlForRevision` and `urlForRevisionRange`

dewei-zhu replied:
For that purpose, we already have `label()`. And changing the behavior of this function may result in a huge refactor as it's used widely.

JonWBedard replied:
I can see why changing `revision` would be a problem, not `urlForRevision`, though.

dewei-zhu replied:
I started with this approach. But in order to do that, `urlForRevision` will need to explicitly call `CommitLog. _repositoryType` or replicate the code. The former approach coupling two modules together. The latter approach is not DRY.
So I ended up with using identifier whenever it's available.

JonWBedard replied:
Can we not just check if the string is an integer? It's not perfect, but if you have a full hash, the likelihood a full hash being entirely numbers is basically impossible.

dewei-zhu replied:
This logic is too much custom and we don't guarantee revision is always integer, it can also be digit only string.

JonWBedard replied:
Right, that's my point. If you're talking to commits.webkit.org and you get an integer or string that is only digits, that's definitely a revision.

pvollan committed:
Block network extension system calls
https://bugs.webkit.org/show_bug.cgi?id=241633
<rdar://73513138>

Reviewed by Sihui Liu.

After moving content filtering to the Network process and fixing an issue where Safari's injected bundle were
making network calls, network extension system calls can be blocked in the WebContent process.

* Source/WebKit/Resources/SandboxProfiles/ios/com.apple.WebKit.WebContent.sb.in:
* Source/WebKit/WebProcess/com.apple.WebProcess.sb.in:

Canonical link: https://commits.webkit.org/251577@main
git-svn-id: https://svn.webkit.org/repository/webkit/trunk@295572 268f45cc-cd09-0410-ab3c-d52691b4dbfc

anttijk committed:
REGRESSION (r287195): Safari fails to correctly render indented numbered lists with custom CSS and hyphenation ON
https://bugs.webkit.org/show_bug.cgi?id=241630
<rdar://91245970>

Reviewed by Darin Adler.

* LayoutTests/fast/text/list-and-hyphenation-expected.html: Added.
* LayoutTests/fast/text/list-and-hyphenation.html: Added.
* Source/WebCore/rendering/RenderText.cpp:
(WebCore::RenderText::maxWordFragmentWidth):

Return entireWordWidth instead of 0 in the bail out cases.
This ends up setting maxWordWidth = w and leaving currMinWidth unchanged in the caller, matching the pre-r287195 behavior.

Canonical link: https://commits.webkit.org/251578@main
git-svn-id: https://svn.webkit.org/repository/webkit/trunk@295573 268f45cc-cd09-0410-ab3c-d52691b4dbfc

patrickangle committed:
Web Inspector: Regression(r293334) `fill-available` is not a valid dimension, `-webkit-fill-available` is still prefixed
https://bugs.webkit.org/show_bug.cgi?id=241649
rdar://95240065

Reviewed by Devin Rousso.

`fill-available` is not a valid CSS dimension, we still need to use `-webkit-fill-available` for now.

* Source/WebInspectorUI/UserInterface/Views/EventBreakpointPopover.css:
(.popover .edit-breakpoint-popover-content .event input):
* Source/WebInspectorUI/UserInterface/Views/GeneralStyleDetailsSidebarPanel.css:
(.sidebar > .panel.details.css-style > .content ~ .options-container > .filter-bar):
* Source/WebInspectorUI/UserInterface/Views/RecordingContentView.css:
(.content-view.recording > .preview-container):

Canonical link: https://commits.webkit.org/251579@main
git-svn-id: https://svn.webkit.org/repository/webkit/trunk@295574 268f45cc-cd09-0410-ab3c-d52691b4dbfc

dcrousso committed:
add myself to GitHub CODEOWNERS

Reviewed by Jonathan Bedard

* .github/CODEOWNERS:
Specifically for:
- Web Inspector (frontend and backend)
- Apple Pay
- Payment Request
- media controls

Canonical link: https://commits.webkit.org/251580@main
git-svn-id: https://svn.webkit.org/repository/webkit/trunk@295575 268f45cc-cd09-0410-ab3c-d52691b4dbfc

Constellation committed:
Add CODEOWNERS file to ping review request automatically to JSC reviewers
https://bugs.webkit.org/show_bug.cgi?id=241058

Reviewed by Jonathan Bedard.

This patch adds JavaScriptCore CODEOWNERS configurations. By using this file, we can request
reviews automatically based on modified files. I added jsc-reviewers group and
this file configures that Source/JavaScriptCore (except for inspector directory)
PR will automatically set jsc-reviewers as a requested reviewer.

* .github/CODEOWNERS: Added.

Canonical link: https://commits.webkit.org/251581@main
git-svn-id: https://svn.webkit.org/repository/webkit/trunk@295576 268f45cc-cd09-0410-ab3c-d52691b4dbfc

emw-apple committed:
[build-webkit] Don't run build-imagediff when cleaning with Xcode
https://bugs.webkit.org/show_bug.cgi?id=241624

Reviewed by Jonathan Bedard.

Fixes "error: Unknown build action ' --clean'" from xcodebuild, as
build-webkit was passing the --clean argument to build-imagediff with an
extra space.

Since the move to XCBuild, cleaning deletes the whole build directory,
so there is nothing left for build-imagediff to clean and we skip this
step altogether.

* Tools/Scripts/build-webkit:

Canonical link: https://commits.webkit.org/251582@main
git-svn-id: https://svn.webkit.org/repository/webkit/trunk@295577 268f45cc-cd09-0410-ab3c-d52691b4dbfc

youssefsoli committed:
[Cocoa] Add feature flag for WebM Experiment
https://bugs.webkit.org/show_bug.cgi?id=241642
<rdar://95228097>

Patch by Youssef Soliman <youssefdevelops@gmail.com> on 2022-06-15
Reviewed by Eric Carlson.

Added internal feature and build flag for experimental WebM player.

* Source/WTF/Scripts/Preferences/WebPreferencesInternal.yaml:
* Source/WTF/wtf/PlatformEnableCocoa.h:

Canonical link: https://commits.webkit.org/251583@main
git-svn-id: https://svn.webkit.org/repository/webkit/trunk@295578 268f45cc-cd09-0410-ab3c-d52691b4dbfc

patrickangle committed:
Add myself (patrickangle) to GitHub CODEOWNERS for Web Inspector

Reviewed by Jonathan Bedard.

* .github/CODEOWNERS:

Canonical link: https://commits.webkit.org/251584@main
git-svn-id: https://svn.webkit.org/repository/webkit/trunk@295579 268f45cc-cd09-0410-ab3c-d52691b4dbfc

patrickangle committed:
Web Inspector: inspector/page/setShowPaintRects.html is a flakey failure since introduction
https://bugs.webkit.org/show_bug.cgi?id=233047
rdar://85345706

Reviewed by Antoine Quint.

Adding test results to the page causes a paint to be scheduled, which can cause paint rects to be requested at times we
are not expecting them as part of the flow of the test. We expect only our explicit change to an element to cause paints
to occur, but currently those paints end up interleaved with paints for adding results to the test page.

To resolve this, we now defer writing results to the test page until the end of each test case. This allows us to wait
to start each test case until there are no paint rects, and to avoid accidently causing a paint rect to appear at
inopportune times for our testing.

* LayoutTests/inspector/page/setShowPaintRects-expected.txt:
* LayoutTests/inspector/page/setShowPaintRects.html:
* LayoutTests/platform/mac/TestExpectations:

Canonical link: https://commits.webkit.org/251585@main
git-svn-id: https://svn.webkit.org/repository/webkit/trunk@295580 268f45cc-cd09-0410-ab3c-d52691b4dbfc

mcatanzaro committed:
[CMake] Check for all required perl modules
https://bugs.webkit.org/show_bug.cgi?id=241648

Patch by Michael Catanzaro <mcatanzaro@redhat.com> on 2022-06-15
Reviewed by Adrian Perez de Castro.

* Source/cmake/WebKitCommon.cmake:

Canonical link: https://commits.webkit.org/251586@main
git-svn-id: https://svn.webkit.org/repository/webkit/trunk@295581 268f45cc-cd09-0410-ab3c-d52691b4dbfc

AndresGonzalezApple committed:
AX ITM: Should not build an isolated tree branch rooted at an object with invalid ID. Should not attempt to remove an object for an invalid ID.
https://bugs.webkit.org/show_bug.cgi?id=241643

Reviewed by Chris Fleizach.

Added check for valid object ID in AXIsolatedTree::collectNodeChangesForSubtree and in updateChildren since in both cases the ID is used as a key for HashMaps which cannot be null.

* Source/WebCore/accessibility/isolatedtree/AXIsolatedTree.cpp:
(WebCore::AXIsolatedTree::collectNodeChangesForSubtree):
(WebCore::AXIsolatedTree::updateChildren):

Canonical link: https://commits.webkit.org/251587@main
git-svn-id: https://svn.webkit.org/repository/webkit/trunk@295582 268f45cc-cd09-0410-ab3c-d52691b4dbfc

djg committed:
Validate descriptor when creating MTLRenderPipelineState
https://bugs.webkit.org/show_bug.cgi?id=241587
rdar://problem/93820440

Patch by Dan Glastonbury <djg@apple.com> on 2022-06-15
Reviewed by Kimmo Kinnunen.

* Source/ThirdParty/ANGLE/src/libANGLE/renderer/metal/mtl_state_cache.mm:
(rx::mtl::ValidateRenderPipelineState):
(rx::mtl::RenderPipelineCache::createRenderPipelineState):
Extend MTlRenderPipelineDescriptor validation to ensure that there is at least
one valid render target set for the the render pipeline. This is required for
certain families of metal devices to avoid a validation failure inside the metal
framework. Moving the failure here will cause the app using ANGLE to return a GL
error instead of crashing the process.

Canonical link: https://commits.webkit.org/251588@main
git-svn-id: https://svn.webkit.org/repository/webkit/trunk@295583 268f45cc-cd09-0410-ab3c-d52691b4dbfc

fujii committed:
Remove unused expandedQuadEdgesInScreenSpace uniform variable
https://bugs.webkit.org/show_bug.cgi?id=241667

Reviewed by Don Olmstead.

expandedQuadEdgesInScreenSpace uniform variable was added by
108289@main (Bug 90308), but unsued since 124039@main (Bug 105377).

* Source/WebCore/platform/graphics/texmap/TextureMapperShaderProgram.h:
Removed it.

Canonical link: https://commits.webkit.org/251589@main
git-svn-id: https://svn.webkit.org/repository/webkit/trunk@295584 268f45cc-cd09-0410-ab3c-d52691b4dbfc

saambarati committed:
Don't clear both prototype ICs for LLInt iterator_next
https://bugs.webkit.org/show_bug.cgi?id=241663

Reviewed by Yusuke Suzuki.

If we use BytecodeIndex instead of a bytecode offset, we don't need to
clear both ICs. We can just clear the one that the watchpoint fired for.

* Source/JavaScriptCore/bytecode/CodeBlock.cpp:
(JSC::CodeBlock::finalizeLLIntInlineCaches):
* Source/JavaScriptCore/bytecode/CodeBlock.h:
* Source/JavaScriptCore/bytecode/LLIntPrototypeLoadAdaptiveStructureWatchpoint.cpp:
(JSC::LLIntPrototypeLoadAdaptiveStructureWatchpoint::LLIntPrototypeLoadAdaptiveStructureWatchpoint):
(JSC::LLIntPrototypeLoadAdaptiveStructureWatchpoint::initialize):
(JSC::LLIntPrototypeLoadAdaptiveStructureWatchpoint::fireInternal):
* Source/JavaScriptCore/bytecode/LLIntPrototypeLoadAdaptiveStructureWatchpoint.h:
* Source/JavaScriptCore/llint/LLIntSlowPaths.cpp:
(JSC::LLInt::setupGetByIdPrototypeCache):
(JSC::LLInt::performLLIntGetByID):
(JSC::LLInt::LLINT_SLOW_PATH_DECL):

Canonical link: https://commits.webkit.org/251590@main
git-svn-id: https://svn.webkit.org/repository/webkit/trunk@295585 268f45cc-cd09-0410-ab3c-d52691b4dbfc

msaboff committed:
[Regression r295565] iOS Simulator build fails
https://bugs.webkit.org/show_bug.cgi?id=241673

Unreviewed, fix ios simulator build after r295565.

* Source/WebKit/Configurations/adattributiond.xcconfig:
* Source/WebKit/Configurations/webpushd.xcconfig:

Canonical link: https://commits.webkit.org/251591@main
git-svn-id: https://svn.webkit.org/repository/webkit/trunk@295586 268f45cc-cd09-0410-ab3c-d52691b4dbfc

fujii committed:
Remove unused code for GL_TEXTURE_RECTANGLE_ARB
https://bugs.webkit.org/show_bug.cgi?id=241670

Reviewed by Don Olmstead.

The code for GL_TEXTURE_RECTANGLE_ARB was added by 105925@main (Bug
87738) for Qt-Mac port. But, no longer used.

* Source/WebCore/platform/graphics/texmap/TextureMapperGL.cpp:
(WebCore::TextureMapperGL::drawTexture):
(WebCore::prepareTransformationMatrixWithFlags):
(WebCore::TextureMapperGL::drawTexturePlanarYUV):
(WebCore::TextureMapperGL::drawTextureSemiPlanarYUV):
(WebCore::TextureMapperGL::drawTexturePackedYUV):
(WebCore::TextureMapperGL::drawTexturedQuadWithProgram):
* Source/WebCore/platform/graphics/texmap/TextureMapperGL.h:
* Source/WebCore/platform/graphics/texmap/TextureMapperShaderProgram.cpp:
(WebCore::TextureMapperShaderProgram::create):
* Source/WebCore/platform/graphics/texmap/TextureMapperShaderProgram.h:

Canonical link: https://commits.webkit.org/251592@main
git-svn-id: https://svn.webkit.org/repository/webkit/trunk@295587 268f45cc-cd09-0410-ab3c-d52691b4dbfc

achristensen07 committed:
TestWebKitAPI.WebKit2.SpeechRecognitionErrorWhenStartingAudioCaptureOnDifferentPageE is a flaky crash
https://bugs.webkit.org/show_bug.cgi?id=239474

Reviewed by Sihui Liu.

At least when using MockAudioSharedInternalUnit the call to RealtimeMediaSource::start can call code on another thread
that assumes that RealtimeMediaSource::setLogger has already been called.  This can be made to crash almost 100% of the time
in the test by adding WTFReportBacktrace at the beginning of RealtimeMediaSource::setLogger which delays when m_logger is set
enough for the "MockAudioSharedInternalUnit Capture Queue" thread to read from m_logger before it is written without this change.

* Source/WebCore/Modules/speech/SpeechRecognitionCaptureSourceImpl.cpp:
(WebCore::SpeechRecognitionCaptureSourceImpl::SpeechRecognitionCaptureSourceImpl):

Canonical link: https://commits.webkit.org/251593@main
git-svn-id: https://svn.webkit.org/repository/webkit/trunk@295588 268f45cc-cd09-0410-ab3c-d52691b4dbfc

youennf committed:
Add WebSocket map size logging
https://bugs.webkit.org/show_bug.cgi?id=241590
rdar://problem/95093425

Patch by Youenn Fablet <youennf@gmail.com> on 2022-06-16
Reviewed by Eric Carlson.

* Source/WebKit/NetworkProcess/cocoa/NetworkSessionCocoa.mm:
(WebKit::NetworkSessionCocoa::addWebSocketTask):
(WebKit::NetworkSessionCocoa::removeWebSocketTask):

Canonical link: https://commits.webkit.org/251594@main
git-svn-id: https://svn.webkit.org/repository/webkit/trunk@295589 268f45cc-cd09-0410-ab3c-d52691b4dbfc

youennf committed:
Moving a media element from one document to another should not break autoplay
https://bugs.webkit.org/show_bug.cgi?id=241631
rdar://95060381

Patch by Youenn Fablet <youennf@gmail.com> on 2022-06-16
Reviewed by Eric Carlson.

When pausing a video element when being detached from a document, autoplay might get broken if the element is interrupted due to invisibility.
In that case, the session will store the fact that the element is paused and when the end of invisibility interruption happens, the session state is Paused instead of Autoplay.
To prevent this, we do not pause when being detached if we are alread interrupted due to invisibility.

Covered by added test.

* LayoutTests/http/tests/webrtc/resources/utility-frame.html: Added.
* LayoutTests/http/tests/webrtc/video-mediastream-invisible-autoplay-detached-expected.txt: Added.
* LayoutTests/http/tests/webrtc/video-mediastream-invisible-autoplay-detached.html: Added.
* Source/WebCore/html/HTMLMediaElement.cpp:
(WebCore::HTMLMediaElement::pauseAfterDetachedTask):

Canonical link: https://commits.webkit.org/251595@main
git-svn-id: https://svn.webkit.org/repository/webkit/trunk@295590 268f45cc-cd09-0410-ab3c-d52691b4dbfc

youennf committed:
A black screen appears in a muted video element outside the viewport
https://bugs.webkit.org/show_bug.cgi?id=241152
rdar://problem/94562636

Patch by Youenn Fablet <youennf@gmail.com> on 2022-06-16
Reviewed by Eric Carlson.

We sometimes remove the invisible autoplay restrictions.
If we remove it while we are suspended due this restriction, we will not unsuspend and will wait for a user interaction.
To prevent this, we do not return early in updateShouldAutoplay if we are interrupted due to the invisible autoplay restriction.

* LayoutTests/fast/mediastream/video-mediastream-restricted-invisible-autoplay-user-click-expected.txt: Added.
* LayoutTests/fast/mediastream/video-mediastream-restricted-invisible-autoplay-user-click.html: Added.
* Source/WebCore/html/HTMLMediaElement.cpp:
(WebCore::HTMLMediaElement::updateShouldAutoplay):

Canonical link: https://commits.webkit.org/251596@main
git-svn-id: https://svn.webkit.org/repository/webkit/trunk@295591 268f45cc-cd09-0410-ab3c-d52691b4dbfc

JonWBedard committed:
[ews-build.webkit.org] Support WebKit/WebKit-security
https://bugs.webkit.org/show_bug.cgi?id=241656
<rdar://problem/95252686>

Reviewed by Ryan Haddad and Aakash Jain.

* Tools/CISupport/ews-build/steps.py:

Canonical link: https://commits.webkit.org/251597@main
git-svn-id: https://svn.webkit.org/repository/webkit/trunk@295592 268f45cc-cd09-0410-ab3c-d52691b4dbfc

AndresGonzalezApple committed:
AX ITM: VoiceOver cannot get contextual menus for links on some pages with ITM enabled.
https://bugs.webkit.org/show_bug.cgi?id=241668

Reviewed by Chris Fleizach.

In isolated tree mode the performSelector method was being called off the main thread, and was not invoking the given selector. This patch fixes the problem by dispatching to the main thread the call to performSelector. Some code cleanup.

* Source/WebCore/accessibility/mac/WebAccessibilityObjectWrapperMac.mm:
(-[WebAccessibilityObjectWrapper accessibilityPerformShowMenuAction]):
(-[WebAccessibilityObjectWrapper _accessibilityShowContextMenu]):
(-[WebAccessibilityObjectWrapper accessibilityShowContextMenu]): Deleted.

Canonical link: https://commits.webkit.org/251598@main
git-svn-id: https://svn.webkit.org/repository/webkit/trunk@295593 268f45cc-cd09-0410-ab3c-d52691b4dbfc

youennf committed:
Tab snapshotting should not trigger the WebGL code path when HAVE(LOW_AV_SAMPLE_BUFFER_PRUNING_INTERVAL) is false
https://bugs.webkit.org/show_bug.cgi?id=241628
rdar://94484791

Patch by Youenn Fablet <youennf@gmail.com> on 2022-06-16
Reviewed by Jer Noble.

If HAVE(LOW_AV_SAMPLE_BUFFER_PRUNING_INTERVAL) is false, we either render the video or we can grab individual video frames for canvas rendering.
We render video except when WebGL canvas painting is used.
Before the patch, the GPUProcess code path was relying on calling MediaPlayerPrivateMediaSourceAVFObjC::videoFrameForCurrentTime for tab snapshotting.
This then prevented the video to be rendered.

This patch is adding a MediaPlayer::willBeAskedToPaintGL method to ask the player to enter the WebGL code path to grab individual video frames.
This method is called when creating a video texture for WebGL.
MediaPlayerPrivateMediaSourceAVFObjC implements this method to switch to the decompression session to grab individual video frames.
Other players are left unchanged.
We add the necessary IPC handling to send the signal from WebProcess to GPUProcess.
All of this is specific to HAVE(LOW_AV_SAMPLE_BUFFER_PRUNING_INTERVAL) being false.

Manually tested by tabsnapshooting youtube pages as well as loading web page using MSE+WebGL.

* Source/WebCore/html/canvas/WebGLRenderingContextBase.cpp:
(WebCore::WebGLRenderingContextBase::texImageSourceHelper):
* Source/WebCore/platform/graphics/MediaPlayer.cpp:
(WebCore::MediaPlayer::willBeAskedToPaintGL):
* Source/WebCore/platform/graphics/MediaPlayer.h:
* Source/WebCore/platform/graphics/MediaPlayerPrivate.h:
(WebCore::MediaPlayerPrivateInterface::willBeAskedToPaintGL):
* Source/WebCore/platform/graphics/avfoundation/objc/MediaPlayerPrivateMediaSourceAVFObjC.h:
* Source/WebCore/platform/graphics/avfoundation/objc/MediaPlayerPrivateMediaSourceAVFObjC.mm:
(WebCore::MediaPlayerPrivateMediaSourceAVFObjC::willBeAskedToPaintGL):
(WebCore::MediaPlayerPrivateMediaSourceAVFObjC::videoFrameForCurrentTime):
(WebCore::MediaPlayerPrivateMediaSourceAVFObjC::shouldEnsureLayer const):
(WebCore::MediaPlayerPrivateMediaSourceAVFObjC::acceleratedRenderingStateChanged):
* Source/WebKit/GPUProcess/media/RemoteMediaPlayerProxy.h:
* Source/WebKit/GPUProcess/media/RemoteMediaPlayerProxy.messages.in:
* Source/WebKit/GPUProcess/media/cocoa/RemoteMediaPlayerProxyCocoa.mm:
(WebKit::RemoteMediaPlayerProxy::willBeAskedToPaintGL):
* Source/WebKit/WebProcess/GPU/media/MediaPlayerPrivateRemote.cpp:
(WebKit::MediaPlayerPrivateRemote::willBeAskedToPaintGL):
* Source/WebKit/WebProcess/GPU/media/MediaPlayerPrivateRemote.h:

Canonical link: https://commits.webkit.org/251599@main
git-svn-id: https://svn.webkit.org/repository/webkit/trunk@295594 268f45cc-cd09-0410-ab3c-d52691b4dbfc

csaavedra committed:
TextureMapper: remove unused size variable in draw texture methods
https://bugs.webkit.org/show_bug.cgi?id=241679

Reviewed by Darin Adler.

WebCore::TextureMapperGL::drawTexturedQuadWithProgram() doesn't really
use the size parameter, remove it from everwhere where it's passed
forward.

* Source/WebCore/platform/graphics/texmap/TextureMapperGL.cpp:
(WebCore::TextureMapperGL::drawTexture):
(WebCore::prepareTransformationMatrixWithFlags):
(WebCore::TextureMapperGL::drawTexturePlanarYUV):
(WebCore::TextureMapperGL::drawTextureSemiPlanarYUV):
(WebCore::TextureMapperGL::drawTexturePackedYUV):
(WebCore::TextureMapperGL::drawTexturedQuadWithProgram):
(WebCore::TextureMapperGL::drawFiltered):
(WebCore::TextureMapperGL::drawTextureExternalOES):
* Source/WebCore/platform/graphics/texmap/TextureMapperGL.h:
* Source/WebCore/platform/graphics/texmap/TextureMapperPlatformLayerBuffer.cpp:
(WebCore::TextureMapperPlatformLayerBuffer::paintToTextureMapper):

Canonical link: https://commits.webkit.org/251600@main
git-svn-id: https://svn.webkit.org/repository/webkit/trunk@295595 268f45cc-cd09-0410-ab3c-d52691b4dbfc

szewai committed:
Regression (r295056): do not delete WebSQL directory until there is no use of it in WebKit clients
https://bugs.webkit.org/show_bug.cgi?id=241671

Reviewed by Youenn Fablet.

We have internal clients that set the custom WebSQL directory do not only use the directory for WebSQL.

* Source/WebKit/UIProcess/WebsiteData/WebsiteDataStore.cpp:
(WebKit::WebsiteDataStore::resolveDirectoriesIfNecessary):

Canonical link: https://commits.webkit.org/251601@main
git-svn-id: https://svn.webkit.org/repository/webkit/trunk@295596 268f45cc-cd09-0410-ab3c-d52691b4dbfc

twilco committed:
AX: Update the isolated tree in response to dynamic placeholder and aria-placeholder changes
https://bugs.webkit.org/show_bug.cgi?id=241675

Reviewed by Chris Fleizach.

* LayoutTests/accessibility/placeholder-expected.txt:
* LayoutTests/accessibility/placeholder.html: Add new test cases.
* LayoutTests/platform/ios/TestExpectations: Enable accessibility/placeholder.html.
* Source/WebCore/accessibility/AXLogger.cpp:
(WebCore::operator<<):
* Source/WebCore/accessibility/AXObjectCache.cpp:
(WebCore::AXObjectCache::handleAttributeChange):
(WebCore::AXObjectCache::updateIsolatedTree):
* Source/WebCore/accessibility/AXObjectCache.h:
* Source/WebCore/accessibility/atspi/AXObjectCacheAtspi.cpp

Canonical link: https://commits.webkit.org/251602@main
git-svn-id: https://svn.webkit.org/repository/webkit/trunk@295597 268f45cc-cd09-0410-ab3c-d52691b4dbfc

briannafan committed:
Extend build-and-collect-pgo-profiles to support testing a browser path
https://bugs.webkit.org/show_bug.cgi?id=241392
rdar://94581810

Patch by briannafan <briannaf@berkeley.edu> on 2022-06-16
Reviewed by Dewei Zhu.

Extending script to take browser path and skip building.

* Tools/Scripts/build-and-collect-pgo-profiles:

Canonical link: https://commits.webkit.org/251603@main
git-svn-id: https://svn.webkit.org/repository/webkit/trunk@295598 268f45cc-cd09-0410-ab3c-d52691b4dbfc

JonWBedard committed:
[Merge-Queue] Validate reviewers in commit message (Follow-up)
https://bugs.webkit.org/show_bug.cgi?id=240718
<rdar://problem/93665771>

Unreviewed infrastructure fix.

* Tools/CISupport/ews-build/steps.py:
(ValidateCommitMessage.run): Allow for unreviewed changes.

Canonical link: https://commits.webkit.org/251604@main
git-svn-id: https://svn.webkit.org/repository/webkit/trunk@295599 268f45cc-cd09-0410-ab3c-d52691b4dbfc

pvollan committed:
Handle display reconfiguration on main thread
https://bugs.webkit.org/show_bug.cgi?id=241683
<rdar://83417447>

Reviewed by Simon Fraser.

It is not safe to handle this on a non main thread, since we send messages to the WebContent and GPU process.

* Source/WebKit/UIProcess/WebProcessPool.cpp:
(WebKit::displayReconfigurationCallBack):

Canonical link: https://commits.webkit.org/251605@main
git-svn-id: https://svn.webkit.org/repository/webkit/trunk@295600 268f45cc-cd09-0410-ab3c-d52691b4dbfc

xeenon committed:
Add xeenon GitHub account to contributors.json.
https://bugs.webkit.org/show_bug.cgi?id=241686

Unreviewed.

* metadata/contributors.json:
Add xeenon GitHub account under Timothy Hatcher.

Canonical link: https://commits.webkit.org/251606@main
git-svn-id: https://svn.webkit.org/repository/webkit/trunk@295601 268f45cc-cd09-0410-ab3c-d52691b4dbfc

shvaikalesh committed:
[WebIDL] maplike<> and setlike<> declarations should be resilient to tampered prototypes
https://bugs.webkit.org/show_bug.cgi?id=241617
<rdar://93229569>

Reviewed by Yusuke Suzuki.

With this change, maplike<> and setlike<> declarations work as expected if methods of
Map.prototype / Set.prototype are removed, as they are suppossed to per spec [1][2].
Usage of backing Map / Set is an implementation detail of WebKit bindings and should
not be observable.

The fix mirrors all Map / Set prototype methods and "size" getter by private names,
which are inacessible to userland code, ensuring that public JSFunction* instances
are reused to avoid memory bloat.

Also, this change:
  * saves creating 4 extra JSFunction* instances during init of Map / Set prototypes;
  * speeds-up call forwarding by retrieving methods from prototypes with getDirect();
  * aligns property order with the spec, even though there is no requirement.

[1] https://webidl.spec.whatwg.org/#es-maplike
[2] https://webidl.spec.whatwg.org/#es-setlike

* LayoutTests/imported/w3c/web-platform-tests/css/css-highlight-api/Highlight-setlike-tampered-Set-prototype-expected.txt: Added.
* LayoutTests/imported/w3c/web-platform-tests/css/css-highlight-api/Highlight-setlike-tampered-Set-prototype.html: Added.
* LayoutTests/imported/w3c/web-platform-tests/css/css-highlight-api/HighlightRegistry-maplike-tampered-Map-prototype-expected.txt: Added.
* LayoutTests/imported/w3c/web-platform-tests/css/css-highlight-api/HighlightRegistry-maplike-tampered-Map-prototype.html: Added.
* Source/JavaScriptCore/DerivedSources-output.xcfilelist:
* Source/JavaScriptCore/DerivedSources.make:
* Source/JavaScriptCore/JavaScriptCore.xcodeproj/project.pbxproj:
* Source/JavaScriptCore/builtins/BuiltinNames.h:
* Source/JavaScriptCore/runtime/MapPrototype.cpp:
(JSC::MapPrototype::finishCreation):
* Source/JavaScriptCore/runtime/SetPrototype.cpp:
(JSC::SetPrototype::finishCreation):
* Source/WebCore/bindings/js/JSDOMBindingInternals.js:
(forEachWrapper):
* Source/WebCore/bindings/js/JSDOMMapLike.cpp:
(WebCore::getBackingMap):
(WebCore::clearBackingMap):
(WebCore::setToBackingMap):
(WebCore::forwardFunctionCallToBackingMap):
* Source/WebCore/bindings/js/JSDOMMapLike.h:
(WebCore::forwardSizeToMapLike):
(WebCore::forwardEntriesToMapLike):
(WebCore::forwardKeysToMapLike):
(WebCore::forwardValuesToMapLike):
(WebCore::forwardClearToMapLike):
(WebCore::forwardGetToMapLike):
(WebCore::forwardHasToMapLike):
(WebCore::forwardSetToMapLike):
(WebCore::forwardDeleteToMapLike):
* Source/WebCore/bindings/js/JSDOMSetLike.cpp:
(WebCore::getBackingSet):
(WebCore::clearBackingSet):
(WebCore::addToBackingSet):
(WebCore::forwardFunctionCallToBackingSet):
* Source/WebCore/bindings/js/JSDOMSetLike.h:
(WebCore::forwardSizeToSetLike):
(WebCore::forwardEntriesToSetLike):
(WebCore::forwardKeysToSetLike):
(WebCore::forwardValuesToSetLike):
(WebCore::forwardClearToSetLike):
(WebCore::forwardHasToSetLike):
(WebCore::forwardAddToSetLike):
(WebCore::forwardDeleteToSetLike):
* Source/WebCore/bindings/js/WebCoreBuiltinNames.h:

Canonical link: https://commits.webkit.org/251607@main
git-svn-id: https://svn.webkit.org/repository/webkit/trunk@295602 268f45cc-cd09-0410-ab3c-d52691b4dbfc

youennf committed:
SharedWorkerGlobalScope::close should not directly stop its thread
https://bugs.webkit.org/show_bug.cgi?id=241677
rdar://95010255

Patch by Youenn Fablet <youennf@gmail.com> on 2022-06-16
Reviewed by Alex Christensen.

We were directly stopping the worker thread when calling SharedWorkerGlobalScope::close.
This can trigger the case of stopping the worker thread concurrently from main thread and shared worker thread.
Instead, we reuse the existing Worker infastructure to tell its SharedWorkerThreadProxy to terminate.

* Source/WebCore/workers/WorkerOrWorkletThread.cpp:
(WebCore::WorkerOrWorkletThread::stop):
* Source/WebCore/workers/shared/SharedWorkerGlobalScope.cpp:
(WebCore::SharedWorkerGlobalScope::close): Deleted.
* Source/WebCore/workers/shared/SharedWorkerGlobalScope.h:
* Source/WebCore/workers/shared/context/SharedWorkerThreadProxy.cpp:
(WebCore::SharedWorkerThreadProxy::workerGlobalScopeClosed):
* Source/WebCore/workers/shared/context/SharedWorkerThreadProxy.h:

Canonical link: https://commits.webkit.org/251608@main
git-svn-id: https://svn.webkit.org/repository/webkit/trunk@295603 268f45cc-cd09-0410-ab3c-d52691b4dbfc

kcheney1 committed:
File size label for attachment in mail compose shows up black instead of expected grey
https://bugs.webkit.org/show_bug.cgi?id=241600
rdar://94071484

Reviewed by Cameron McCormack.

As of https://trac.webkit.org/changeset/293867/webkit in DrawGlyphsRecorder
specifically, we use the fill color of the initial state of GraphicsContext
in the case where the new fill color we are updating to is equal to the existing
fill color for perf reasons. This revealed a bug where CGContext state gets
out of sync with it's wrapping GraphicsContext, causing an incorrect
initial state fill color and thus painting the second line of Mail
attachments in the wrong color.

This patch fixes this bug by saving and restoring the state of
m_internalContext, which we reference in the case above to determine
whether to update the fill color.

* Source/WebCore/platform/graphics/coretext/DrawGlyphsRecorderCoreText.cpp:
(WebCore::DrawGlyphsRecorder::drawNativeText):

Canonical link: https://commits.webkit.org/251609@main
git-svn-id: https://svn.webkit.org/repository/webkit/trunk@295604 268f45cc-cd09-0410-ab3c-d52691b4dbfc

emw-apple committed:
Add lldbWebKitTester to WebKit.xcworkspace

Unreviewed test fix.

When we switched build-webkit to workspace builds in
https://commits.webkit.org/251501@main, we stopped building it.

lldbWebKitTester must only build on macOS. Add a "lldbWebKitTester
(Platform filters)" aggregate target that selectively builds
lldbWebKitTester on Mac only, like we did for MiniBrowser in
https://commits.webkit.org/251501@main. Update the relevant schemes to
build that aggregate.

We cannot rely on lldbWebKitTester's "SUPPORTED_PLATFORMS = macosx" as a
way to make it always build for Mac, because it depends on WTF and
therefore must only build if WTF is also being built for Mac.

* Tools/lldb/lldbWebKitTester/Configurations/Base.xcconfig: Claim to
  support every platform, so that the aggregate target is always
  buildable and able to apply a platform filter.
* Tools/lldb/lldbWebKitTester/Configurations/lldbWebKitTester.xcconfig:
  Move "SUPPORTED_PLATFORMS = macosx" here, because lldbWebKitTester
  really does need to be restricted to building for Mac.

* Tools/lldb/lldbWebKitTester/lldbWebKitTester.xcodeproj/project.pbxproj:
* WebKit.xcworkspace/xcshareddata/xcschemes/All Modules.xcscheme:
* WebKit.xcworkspace/xcshareddata/xcschemes/All Tools.xcscheme:

Canonical link: https://commits.webkit.org/251610@main
git-svn-id: https://svn.webkit.org/repository/webkit/trunk@295605 268f45cc-cd09-0410-ab3c-d52691b4dbfc

saambarati committed:
Change how we rewind instructions in Air's O0 register allocator
https://bugs.webkit.org/show_bug.cgi?id=241687

Reviewed by Mark Lam.

* Source/JavaScriptCore/b3/air/AirAllocateRegistersAndStackAndGenerateCode.cpp:
(JSC::B3::Air::GenerateAndAllocateRegisters::generate):

Canonical link: https://commits.webkit.org/251611@main
git-svn-id: https://svn.webkit.org/repository/webkit/trunk@295606 268f45cc-cd09-0410-ab3c-d52691b4dbfc

megangardner committed:
Implement alternate fullscreen controls.
https://bugs.webkit.org/show_bug.cgi?id=241654

Reviewed by Tim Horton.

Alternative UI for fullscreen video controls.

* Source/WTF/Scripts/Preferences/WebPreferences.yaml:
* Source/WebKit/UIProcess/ios/fullscreen/WKFullScreenViewController.mm:
(-[WKFullScreenViewController loadView]):

Canonical link: https://commits.webkit.org/251612@main
git-svn-id: https://svn.webkit.org/repository/webkit/trunk@295607 268f45cc-cd09-0410-ab3c-d52691b4dbfc

shvaikalesh committed:
CommonSlowPaths::putDirectWithReify() is incorrect for DontDelete properties
https://bugs.webkit.org/show_bug.cgi?id=241651
<rdar://94016559>

Reviewed by Saam Barati.

Provided the base object has no read-only / accessor / custom properties,
putDirectWithReify() is incorrect for an object with non-configurable property,
whether it's on the structure or a non-reified static none.

In that case, putDirectWithReify() ignores existing non-configurable / non-reified
descriptor and produces an incorrect property descriptor instead of throwing TypeError.
One it's observed in the wild is via an instance field [1].

The issue was due to incorrect ReadOnly-focused check for putDirect() fast path,
which would be correct for [[Set]] but not for [[DefineOwnProperty]].

Apart from introducing tighter fast patch check, this change extracts
JSFunction::mayHaveNonReifiedPrototype() helper, cleaning up JSFunction's overrides,
and removes now unused PutModeDefineOwnPropertyIgnoringExtensibility, which apart from
ignoring extensibility, also ignored read-only / accessor / custom properties,
which felt a bit counter-intuitive.

This change carefully preserves the fast path introduced in webkit.org/b/232479.

[1]: https://tc39.es/ecma262/#sec-definefield

* Source/JavaScriptCore/runtime/CommonSlowPaths.h:
(JSC::CommonSlowPaths::originalStructureBeforePut):
(JSC::CommonSlowPaths::canPutDirectFast):
(JSC::CommonSlowPaths::putDirectWithReify):
(JSC::CommonSlowPaths::putDirectAccessorWithReify):
* Source/JavaScriptCore/runtime/JSFunction.cpp:
(JSC::JSFunction::getOwnPropertySlot):
(JSC::JSFunction::put):
(JSC::JSFunction::deleteProperty):
(JSC::JSFunction::defineOwnProperty):
(JSC::JSFunction::reifyLazyPropertyIfNeeded):
(JSC::JSFunction::reifyLazyPrototypeIfNeeded):
* Source/JavaScriptCore/runtime/JSFunction.h:
* Source/JavaScriptCore/runtime/JSFunctionInlines.h:
(JSC::JSFunction::mayHaveNonReifiedPrototype):
* Source/JavaScriptCore/runtime/JSObject.cpp:
(JSC::JSObject::putDirectCustomAccessor):
(JSC::JSObject::putDirectNonIndexAccessor):
* Source/JavaScriptCore/runtime/JSObject.h:
(JSC::JSObject::putDirect):
(JSC::JSObject::putDirectRespectingExtensibility): Deleted.
* Source/JavaScriptCore/runtime/JSObjectInlines.h:
(JSC::JSObject::putDirectInternal):
* Source/JavaScriptCore/tools/JSDollarVM.cpp:
(JSC::JSC_DEFINE_HOST_FUNCTION):
(JSC::JSDollarVM::finishCreation):

Canonical link: https://commits.webkit.org/251613@main
git-svn-id: https://svn.webkit.org/repository/webkit/trunk@295608 268f45cc-cd09-0410-ab3c-d52691b4dbfc

dcrousso committed:
Web Inspector: use proper name for ITP debug mode
https://bugs.webkit.org/show_bug.cgi?id=241697

Reviewed by Patrick Angle.

* Source/WebInspectorUI/UserInterface/Base/Main.js:
(WI._handleDeviceSettingsTabBarButtonClicked):

* Source/WebInspectorUI/Localizations/en.lproj/localizedStrings.js:

Canonical link: https://commits.webkit.org/251614@main
git-svn-id: https://svn.webkit.org/repository/webkit/trunk@295609 268f45cc-cd09-0410-ab3c-d52691b4dbfc

shvaikalesh committed:
AbsenceOfSetEffect property condition should mind put() overrides
https://bugs.webkit.org/show_bug.cgi?id=241574
<rdar://91833733>

Reviewed by Yusuke Suzuki.

Since JSArray's "length" and RegExpObject's "lastIndex" may be reconfigured as non-writable,
we need to handle them separately in AbsenceOfSetEffect property condition to ensure that compiler
takes a slow path in that case, following the spec and throwing an exception in strict mode [1].

I'm not sure how to make a test case capturing this though.

[1]: https://tc39.es/ecma262/#sec-ordinarysetwithowndescriptor (step 2.a)

* Source/JavaScriptCore/bytecode/PropertyCondition.cpp:
(JSC::nonStructurePropertyMayBecomeReadOnlyWithoutTransition):
(JSC::PropertyCondition::isStillValidAssumingImpurePropertyWatchpoint const):

Canonical link: https://commits.webkit.org/251615@main
git-svn-id: https://svn.webkit.org/repository/webkit/trunk@295610 268f45cc-cd09-0410-ab3c-d52691b4dbfc

dewei-zhu committed:
Prefer using commit identifier to build commit URL.
https://bugs.webkit.org/show_bug.cgi?id=241646
rdar://93104485

Reviewed by Jonathan Bedard.

Use commit identifier to construct commit URL and blame URL when available.
This simplifies URL construction for Git transition.

* Websites/perf.webkit.org/public/v3/models/commit-log.js: Added code to prefer using commit identifiers
when available.
(CommitLog.prototype.url):
(CommitLog.prototype.diff):
* Websites/perf.webkit.org/unit-tests/commit-log-tests.js: Added a unit test for diff.
(assert.deepStrictEqual.webkitGitCommitWithRevisionIdentifier.diff.oldWebKitGitCommitWithRevisionIdentifier):
* Websites/perf.webkit.org/unit-tests/commit-set-tests.js: Updated unit test.
* Websites/perf.webkit.org/unit-tests/resources/mock-v3-models.js: Added 'blameUrl' to 'WebKitGit' mock repository.
(MockModels.inject):

Canonical link: https://commits.webkit.org/251616@main
git-svn-id: https://svn.webkit.org/repository/webkit/trunk@295611 268f45cc-cd09-0410-ab3c-d52691b4dbfc
